### ISSUE TITLE
Zjednodušení použití nastavení pomocí "Datacolor Spyder Checkr"

### ISSUE THREAD

**Issue by Pepa-D750** [darktable-org/darktable#16137]
I have a question, whether it would be possible when setting the colors using "Datacolor Spyder Checkr" in the color calibration module - calibrate using the eyedropper - after setting the corners of the table and calculating the correct values - click the reset button to calculate the profile. It is not possible to do just one press and the tuning between the reading of these values in this module and the adjustment in the exposure module should not be done manually as it is now. Now I press in the calibration module, calculate the values for setting the exposure module. I go to the exposure module and set it there. I return to the calibration module, calculate again and set the new values again in the exposure module, and this sometimes has to be done more than ten times before the exposure adjustment values do not change. This could certainly be done automatically, these are just calculations that can pass the correct value to each other. It seems to me that human activity is somewhat unnecessary here. But maybe there is no other way. What do you think? What is your view on this matter?
Thank you and have a nice day.
Joseph

**Alternatives**

**Additional context**

**Comment by ralfbrown** [comment]
Please fill out the requestetd information, particularly the version of darktable you are using.  The earliest version that had this functionality gave relative adjustments, which resulted in exactly the sort of back-and-forth you report.  But that has since been changed to show the value to be entered in the exposure module, which dramatically reduces this (any remaining changes after the first calibration and adjustment should be very small and decrease extremely quickly with further repetitions).  Due to rounding of the displayed results, it might never get to a set of values which won't change at all on re-calibrating -- if only the last displayed digit is changing, you should consider the process complete.

**Comment by Pepa-D750** [comment]
darktable version
4.6.0

What OS are you using?
Linux

What is the version of your OS?
openSUSE Leap 15.5

Describe your system?
OS: openSUSE Leap 15.5 x86_64  
Kernel: 5.14.21-150500.55.39-default  
Shell: bash 4.4.23  
Resolution: 2560x1440  
DE: Plasma 5.27.9  
WM: KWin  
CPU: AMD Phenom II X4 840 (4) @ 3.200GHz [16.5°C]  
GPU: NVIDIA GeForce GT 730  
Memory: 1.94GiB / 11.67GiB (16%) 

Are you using OpenCL GPU in darktable?
Yes

If yes, what is the GPU card and driver?
No response

**Comment by MStraeten** [comment]
> This could certainly be done automatically, these are just calculations that can pass the correct value to each other.

that’s true but not in any cases. The pixelpipe between exposure and colorcalibration can contain further modules changing the tonal values so it depends on the workflow of the user if such function make sense or messes up stuff.
Additionally darktable module in general doesn’t change other modules parameters. You might do this by custom lua scripts

**Comment by todd-prior** [comment]
I feel like the function when it originally came in presented a delta exposure so you had to do the math and correct from your current exposure. Then at some point AP changed this to a value that reflected the exposure value that should be set to make the values accurate as calculated (as-is was added to the manual). No iterations to shrink this were ever mentioned and if changing exposure impacts some intermediate modules you could keep chasing your tail with minimal gain... In fact AP or the authour of the manual goes so far as to call this an optional step  

"This step is optional and is useful only when the utmost precision is required, but beware that it can produce negative RGB values that will be clipped at various points in the pipeline."

When I use this I have the bare minimum of modules possible and set the color. I find the models for none, neutral, average and on occasion max will give the lowest deltaE generally. I set the exposure from the white patch if needed before the correction and usually find not a big change is calculated for the exposure normalization
